### ISSUE TITLE
facter fact rescue not functional

### ISSUE THREAD

**Issue by rismoney** [puppet-community/puppet-iis#61]
iis_version facter fact yields-

fyi - @cyberious 

```powershell

iis_version => Get-ItemProperty : Cannot find path 'HKLM:\SOFTWARE\Microsoft\InetStp\' because it does not exist.
At line:1 char:2
+ (Get-ItemProperty HKLM:\SOFTWARE\Microsoft\InetStp\ -Name VersionString).Version ...
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (HKLM:\SOFTWARE\Microsoft\InetStp\:String) [Get-ItemProperty], ItemNotFo
   undException
    + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.GetItemPropertyCommand

You cannot call a method on a null-valued expression.
At line:1 char:1
+ (Get-ItemProperty HKLM:\SOFTWARE\Microsoft\InetStp\ -Name VersionString).Version ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull
```

I think this is because the rescue is not being invoked.  you may need to use something like -erroraction -silentlycontinue for the key not found handling.

**Comment by cyberious** [comment]
Thanks @rismoney what version of windows? What version of facter? is it the CFacter 3.x releases or is it 2.x?

**Comment by rismoney** [comment]
facter 2.0.2
windows 2012r2

**Comment by cyberious** [comment]
is IIS not installed?

**Comment by rismoney** [comment]
in the test env I saw this on, no, iis- but I did not examine the whole env since we have a slew of both web and non web infrastructure.

**Comment by rismoney** [comment]
$regkey = Get-ItemProperty HKLM:\SOFTWARE\Microsoft\InetStp\ -Name VersionString -ea silentlyc
ontinue;if ($regkey) {$regkey.VersionString.SubString(8,3)}
                                                           î

**Comment by cyberious** [comment]
Closed with #62
